db.go

```diff
@@ -267,17 +267,9 @@ func (db *DB) run() {
 		case <-db.compactc:
 			db.metrics.compactionsTriggered.Inc()
 
-			_, err1 := db.retentionCutoff()
```

Comment:
We're now relying on compact() to call reload() and thus delete old data, which only happens after compaction. We probably need to call it more often.

Reply:
I thought about this too. But then I realized that our retention decision is based on the most recent block. So effectively nothing can change about the retention state unless `compact()` did something.

Thus it is sufficient.

Reply:
I'm not convinced. Time can move forward, and a user may have (inadvisably) disabled compaction.

Reply:
Certainly worth a doc comment though.

Reply:
Time can move forward but TSDB has nothing to do with wall time at all. Everything is purely based on timestamps of inserted data.
Adding wall time would come with a ton of complications and would make TSDB unsuitable for any data that's not timestamped within a very close range to "real world now".

That was actually my motivation once to have Prometheus continuously write a single series with the current wall timestamp into TSDB in the background. To enforce the coupling.

Reply:
That makes sense.